src/app/components/validate-new-password/ValidateNewPassword.jsx

```diff
@@ -155,6 +156,7 @@ export class ValidateNewPassword extends Component {
                     onChange={this.handleInputChange}
                     disableShowPasswordText={true}
```

Comment:
I think you do not need so complex state. the type of this input is always the same. can you please move it form state to `type='password' ` here. I think this will help simplify the state. It is important to keep the minimum possible in state.

Reply:
The state does change based on if the user clicks show password, By default is of type password so that the password remains hidden. When the user clicks show password, the type changes to text

Reply:
oh, that make sense, but maybe flatten the state

Reply:
can you please change the event on checkbox form handleInputChange to TogglePasswordVisibility and only change the type in state , nothing else. keep it simple. :-) Functions needs to be pure and do only one thing at the time :-)